### ISSUE TITLE
User profiles directory structure

### ISSUE THREAD

**Issue by cipharius** [divnix/digga#312]
## Would your feature fix an existing issue?
Didn't run into others bringing it up, but I believe that user profiles can be more consistent with system profiles and simpler to manage.

## Describe the solution you'd like
Move the user profiles out of the "users" directory, as right now a whole configuration category is resting aside system user definitions. As a newcommer I found this confusing. I see it being equivalent as placing "profiles" under "hosts", just because that directory is concerned with system level configurations.

I imagine having the user profiles be located next to the system profiles. Either in the top directory, or have "system" and "user" directories under a top level "profiles" directory.

## Describe alternatives you've considered
At least document the reasoning behind the current directory structure, to make it easier to get onboard with DevOS mindset.

## Additional context
Important semantics to newcommers.

**Comment by blaggacao** [comment]
Reducing the top level `tree -L1` (and also probably `tree -L2` footprint is a worthwhile goal, we adscribe ourselves to.

I can connect with your argument (expanded for home manager):

```
./profiles/system
./profiles/users
./profiles/hm # from ./users/profiles

./modules/hm  # from ./users/modules
```

What do @nrdxp & @Pacman99 think?

**Comment by cipharius** [comment]
So you mentioned seperate `users` and `hm` profile directories. What exactly would they contain? 

Is `users` meant for system users definition(like `./users`) or is it the vanilla NixOS user profiles, as opposed to home-manager profiles?

**Comment by nrdxp** [comment]
So the current `users` directory is essentially the equivalent of hosts for home-manager, but now that you mention it, it doesn't really make sense to have the profiles there, and I can see how that'd be confusing. I did used to have a symlink in the tree from `profiles/users` to `users/profiles`, so maybe we could just move `users/profiles` permanently under profiles instead.

With that said, I'm not 100% sure the entire users directory should live under profiles. Maybe `hosts/users` or something to that effect?

**Comment by cipharius** [comment]
That clears up some things.

If `users` is pretty much equivalent to the `hosts`, except for user definition, why each user has dedicated directory while hosts suffice with just a file matching the hostname? I think following the hosts style would make the intended purpose of that directory clearer. Otherwise now the user directories imply that something else might be placed there, not just the `default.nix` user's definition.

From the discussion it seems that this kind of directory tree would be simpler to understand and use:
```
hosts/
    NixOS.nix
users/
    nixos.nix
    root.nix
profiles/
    cachix/
        ...
    core/
        ...
    users/  # from users/profiles
        ...
modules/
    customBuilds.nix
    module-list.nix
    users/ # from users/modules
        module-list.nix
```

What do you think?

**Comment by Pacman99** [comment]
Not arguing for this to be the devos default, but I do something a little different in my config.
I have a `system` and `home` top-level folder for all the relevant profiles/modules for each configuration.

But I actually think a `profiles/users` and `profiles/modules` is more confusing as a default and that means the `rakeLeaves` call in `nixos` will collect all the user profiles(not that a big of a deal since their aggregated into suites).
Overall I still prefer `users/profiles` and `users/modules` because it shows a clear line of separation between home-manager profiles and nixos profiles. And since the recommendation is to set home-manager things in your user profile, I feel like it naturally makes sense for hm profiles and modules to be set in the same folder.

**Comment by cipharius** [comment]
I also find it interesting how each of you choose slightly different configuration layout.

I'm still trying to get a stable base installation and migrate my previous configuration little by little, so I didn't feel confident enough to rearrange things just yet. Just by looking at core, community and nrdxp branches, the small structural differences left me bit confused how to get started and which practices I should follow.

**Comment by nrdxp** [comment]
> why each user has dedicated directory while hosts suffice with just a file matching the hostname?
I believe with the latest digga develop, you could do either or. I am currently using a `hosts/system/default.nix` approach in my own config

**Comment by Pacman99** [comment]
> I also find it interesting how each of you choose slightly different configuration layout.

I even have differences between two nixos config repo. My server doesn't use home-manager at all so I just dropped the whole users folder and `home` namespace.
I realize its confusing for people to have different structural layouts, but I think its overall a good thing to allow such customization. Every person has different needs in their repositories. I recommend just using core as a base point. Community is planned to be removed at some point.

**Comment by cipharius** [comment]
I wanted to mention that I have managed to migrate over to devos.

Rake leaves function is indeed flexible enough to let me restructure users definitions to match the hosts, or vice versa. For now I did not reposition the `./users/profiles`, as turning user definitions into files made the `modules` and `profiles` directories stand out.

Right now I settled with following structure:

```
hosts/
  host1.nix
  host2.nix
users/
  profiles/
    ...
  modules/
    ...
  user1.nix
  user2.nix
profiles/
  ...
modules/
  ...
```

Not sure what to do with this issue as currently it's not conclusive which would be the cleanest directory layout and the ticket has no clear goal. Each of you has slightly different setup, so I suppose this is something that needs input and discussion from wider community to settle with one layout that most would be satisfied with.
